spec/operators/mergeScan-spec.ts

```diff
@@ -1,140 +1,163 @@
-    const e1 = hot('--a--^--b--c--d--e--f--g--|');
-    const e1subs =      '^               !     ';
-    const expected =    '--------u--v--w--     ';
-
-    const values = {
-      u: ['b'],
-      v: ['c'],
-      w: ['b', 'd'],
-      x: ['c', 'e'],
-      y: ['b', 'd', 'f'],
-      z: ['c', 'e', 'g'],
-    };
-
-    const source = e1.pipe(mergeScan((acc, x) =>
-      of(acc.concat(x)).pipe(delay(50, rxTestScheduler)), [] as string[]));
-
-    expectObservable(source, e1subs).toBe(expected, values);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    testScheduler.run(({ hot, time, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('--a--^--b--c--d--e--f--g--|');
+      const e1subs = '     ^---------------!     ';
+      const t = time('        -----|'); // acc = []; x = 'b'; acc.concat(x) = ['b']; t = 5
+      //                         -----|    acc = []; x = 'c'; acc.concat(x) = ['c']
+      //                            -----| acc = ['b']; x = 'd'; acc.concat(x) = ['b', 'd']
+      const expected = '   --------u--v--w--     ';
+      const unsub = '      ----------------!     ';
+
+      const values = {
+        u: ['b'],
+        v: ['c'],
+        w: ['b', 'd'],
+      };
+
+      const result = e1.pipe(mergeScan((acc, x) => of(acc.concat(x)).pipe(delay(t)), [] as string[]));
+
+      expectObservable(result, unsub).toBe(expected, values);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
   it('should not break unsubscription chains when result is unsubscribed explicitly', () => {
-    const e1 = hot('--a--^--b--c--d--e--f--g--|');
-    const e1subs =      '^               !     ';
-    const expected =    '--------u--v--w--     ';
-    const unsub =       '                !     ';
-
-    const values = {
-      u: ['b'],
-      v: ['c'],
-      w: ['b', 'd'],
-      x: ['c', 'e'],
-      y: ['b', 'd', 'f'],
-      z: ['c', 'e', 'g'],
-    };
-
-    const source = e1
-      .pipe(
+    testScheduler.run(({ hot, time, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('--a--^--b--c--d--e--f--g--|');
+      const e1subs = '     ^---------------!     ';
+      const t = time('        -----|'); // acc = []; x = 'b'; acc.concat(x) = ['b']; t = 5
+      //                         -----|    acc = []; x = 'c'; acc.concat(x) = ['c']
+      //                            -----| acc = ['b']; x = 'd'; acc.concat(x) = ['b', 'd']
+      const expected = '   --------u--v--w--     ';
+      const unsub = '      ----------------!     ';
+
+      const values = {
+        u: ['b'],
+        v: ['c'],
+        w: ['b', 'd'],
+      };
+
+      const result = e1.pipe(
         mergeMap((x) => of(x)),
-        mergeScan((acc, x) =>
-          of([...acc, x]).pipe(
-            delay(50, rxTestScheduler)
-          )
-        , [] as string[]),
-        mergeMap(function (x) { return of(x); })
+        mergeScan((acc, x) => of(acc.concat(x)).pipe(delay(t)), [] as string[]),
```

Comment:
Should I replace this `of` call with the `cold`?

Reply:
Nope, I don't think we should replace any non-`TestScheduler` observables with either `hot` or `cold` because there are some scenarios that they cannot represent. Let's just play it safe.

Reply:
Yeah, smells like #5523. But, how safe was this: https://github.com/ReactiveX/rxjs/pull/5805/files#diff-a76406e2b4935afce22a4ba7e0b415b12350464291e64e81daf12a3b180c7606L326 ?

Reply:
I guess that `EMPTY` doesn't notifies anything upon subscription, so this was safe enough...

Reply:
But, I'm still not sure that in the case of `of(...).pipe(delay(...))` we could have something like what's described in the #5523. The `delay` should not allow synchronous-upon-subscription notifications.
I don't think we _have to_ change anything here now, but I'd just like to be sure that this is not the synchronous-upon-subscription case and that it _should_ be safe to convert this to `cold`/`hot`.
cc @cartant

Reply:
> I guess that, because `EMPTY` doesn't notify anything upon subscription, this was safe enough

`EMPTY` does emit a `complete` notification upon subscription, so it cannot be represented with a `TestScheduler` observable. In the test that you've linked - and as you've mentioned - the use of `delay` - means that the effect will be the same - whether a marble observable or `of` is used. There are going to be cases where the replacement it safe, I just don't want replacements to be made in PRs that have descriptions like 'convert X tests to run mode'.

Reply:
Yes, I understand. That's why I didn't do it in this PR in the first place, but I asked a question should (and could) I replace it later. I won't create any additional PRs for now.